### PR TITLE
[16.0][MIG] hr_employee_relative: Migration to 16.0.

### PR DESCRIPTION
**Pull request by BT-anieto**

### CODE REVIEW

**Comment by BT-anieto** [comment]
@pedrobaeza  I created this PR for hr_employee_relative.

File: requirements.txt
Comment (BT-rmartin) [inline requirements.txt:1]: @pedrobaeza Is this auto-generated? How?

File: hr_employee_relative/views/hr_employee_relative.xml
Comment (BT-rmartin) [inline hr_employee_relative/views/hr_employee_relative.xml]: No need for readonly, a non-stored computed field is now by default readonly

File: hr_employee_relative/views/hr_employee_relative.xml
Comment (BT-rmartin) [inline hr_employee_relative/views/hr_employee_relative.xml:1]: XML header is not needed

File: hr_employee_relative/tests/test_hr_employee_relatives.py
Comment (BT-rmartin) [inline hr_employee_relative/tests/test_hr_employee_relatives.py]: Is there really a need for this? It's just reused twice and the legibility is much worse in the self.env.ref
In fact you could load at the beginning of the test self.env.ref("hr_employee_relative.relation_sibling") and reuse it in lines 28 and 48

File: hr_employee_relative/tests/test_hr_employee_relatives.py
Comment (BT-rmartin) [inline hr_employee_relative/tests/test_hr_employee_relatives.py]: Is this really needed?
When using Form in line 46 it will try to open any editable view of hr.employee.relative. And we have defined the tree view as editable, and it has partner_id and relation_id
No need at all then to do this. By default it will load that view and will be able to apply the onchanges there. 

with Form(relative) as f: should be enough

File: hr_employee_relative/tests/test_hr_employee_relatives.py
Comment (BT-rmartin) [inline hr_employee_relative/tests/test_hr_employee_relatives.py]: This is also not needed as when you exit from with the changes are already saved

File: hr_employee_relative/models/hr_employee_relative.py
Comment (BT-rmartin) [inline hr_employee_relative/models/hr_employee_relative.py]: No need for string

File: hr_employee_relative/models/hr_employee_relative.py
Comment (BT-rmartin) [inline hr_employee_relative/models/hr_employee_relative.py]: No need for string and I would also remove the comodel_name, just giving as first parameter the model value

File: hr_employee_relative/models/hr_employee_relative.py
Comment (BT-rmartin) [inline hr_employee_relative/models/hr_employee_relative.py]: No need for string

File: hr_employee_relative/models/hr_employee_relative.py
Comment (BT-rmartin) [inline hr_employee_relative/models/hr_employee_relative.py:18]: No need for string

File: hr_employee_relative/models/hr_employee.py
Comment (BT-rmartin) [inline hr_employee_relative/models/hr_employee.py:10]: No need for string

File: hr_employee_relative/i18n/de.po
Comment (BT-rmartin) [inline hr_employee_relative/i18n/de.po:7]: @pedrobaeza Not sure how the translations work now with Weblate, if these po files need to be here or they should be removed and once the translation is in Weblate they are automatically merged here. 
Should the person doing the migration do the entries in Weblate?

File: hr_employee_relative/data/data_relative_relation.xml
Comment (BT-rmartin) [inline hr_employee_relative/data/data_relative_relation.xml:1]: XML header not needed

File: hr_employee_relative/README.rst
Comment (BT-rmartin) [inline hr_employee_relative/README.rst:1]: @pedrobaeza the README file is auto-generated when merging. Should it be included in the migration and adapted or just removed so that later it can be added automatically?

**Review by BT-rmartin — CHANGES_REQUESTED** [review]
@BT-anieto Good job. I still think the code can be improved in terms of legibility with some small changes. Would you mind to do them?

File: requirements.txt
Comment (pedrobaeza) [reply requirements.txt:1]: With pre-commit

File: hr_employee_relative/i18n/de.po
Comment (pedrobaeza) [reply hr_employee_relative/i18n/de.po:7]: Yes, they should stay.

File: hr_employee_relative/data/data_relative_relation.xml
Comment (pedrobaeza) [reply hr_employee_relative/data/data_relative_relation.xml:1]: That's not true. Please don't remove them.

File: hr_employee_relative/README.rst
Comment (pedrobaeza) [reply hr_employee_relative/README.rst:1]: Don't touch it to reduce the diff.

File: hr_employee_relative/data/data_relative_relation.xml
Comment (BT-rmartin) [reply hr_employee_relative/data/data_relative_relation.xml:1]: Why is it still needed?

Comment (pedrobaeza) [reply hr_employee_relative/data/data_relative_relation.xml:1]: Because the full XML compliant is with this header. Having `lxml` as current processing library that doesn't force to use it doesn't mean that is the correct thing. If tomorrow the library is changed (or a new version enforces it), we will need to reintroduce it again. Even more, your principle in migration should be to reduce the diff at maximum, unless the change means a good improvement. And the header also specifies the encoding, which is valuable information in case of weird characters (if some texts are put directly in other non-English language, which is not the case here, but imaging in localization repositories).

**Comment by luistorresm** [comment]
Hello @BT-anieto 

Please apply this task to your MR: (You can see this [here](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0))

`Squash administrative commits (if any) with the previous commit for reducing commit noise. They are named as "[UPD] README.rst", "[UPD] Update $MODULE.pot", "Update translation files" and similar names, and comes from OCA-git-bot, oca-travis or oca-transbot. IMPORTANT: Don't squash legit translation commits, authored by their translators, with the message "Translated using Weblate (...)".`

File: hr_employee_relative/data/data_relative_relation.xml
Comment (BT-rmartin) [reply hr_employee_relative/data/data_relative_relation.xml:1]: thanks for clarifying it. Regarding reducing the diff at maximum, then my suggestions of improvement shouldn't have been done in this PR but in a different one once the migration is merged?

**Comment by BT-anieto** [comment]
@luistorresm Thanks for your review! Is my rebase okay? If it is, I will do it for the other two open PRs

**Review by luistorresm — APPROVED** [review]

File: hr_employee_relative/data/data_relative_relation.xml
Comment (pedrobaeza) [reply hr_employee_relative/data/data_relative_relation.xml:1]: Cosmetic changes on others' work should be usually prevented, unless you do a big refactorization that touches most part of such code.

**Review by BT-rmartin — APPROVED** [review]
LGTM. Good job!

**Review by hieulucky111 — APPROVED** [review]
LGTM
Code review

**Review by nimarosa — APPROVED** [review]
Code review looks good.

**Pull request merged**
